client/client_update_test.go

```diff
@@ -28,109 +28,46 @@ func newBlankRepo(t *testing.T, url string) *NotaryRepository {
 	return repo
 }
 
-// If there's no local cache, we go immediately to check the remote server for
-// root, and if it doesn't exist, we return ErrRepositoryNotExist. This happens
-// with or without a force check (update for write).
-func TestUpdateNotExistNoLocalCache(t *testing.T) {
-	testUpdateNotExistNoLocalCache(t, false)
-	testUpdateNotExistNoLocalCache(t, true)
-}
-
-func testUpdateNotExistNoLocalCache(t *testing.T, forWrite bool) {
-	ts, _, _ := simpleTestServer(t)
-	defer ts.Close()
-
-	repo := newBlankRepo(t, ts.URL)
-	defer os.RemoveAll(repo.baseDir)
-
-	// there is no metadata at all - this is a fresh repo, and the server isn't
-	// aware of the root.
-	_, err := repo.Update(forWrite)
-	require.Error(t, err)
-	require.IsType(t, ErrRepositoryNotExist{}, err)
-}
-
-// If there is a local cache, we use the local root as the trust anchor and we
-// then an update. If the server has no root.json, we return an ErrRepositoryNotExist.
-// If we force check (update for write), then it hits the server first, and
-// still returns an ErrRepositoryNotExist.
-func TestUpdateNotExistWithLocalCache(t *testing.T) {
-	testUpdateNotExistWithLocalCache(t, false)
-	testUpdateNotExistWithLocalCache(t, true)
-}
-
-func testUpdateNotExistWithLocalCache(t *testing.T, forWrite bool) {
-	ts, _, _ := simpleTestServer(t)
-	defer ts.Close()
+var metadataDelegations = []string{"targets/a", "targets/a/b", "targets/a/b/c"}
```

Comment:
Would be good to have a `target/b` too just so we have more than a single linear chain.